pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,182 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package djm contains data-jobs-monitoring installation logic
+package djm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/fleet/installer/setup/common"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+const (
+	emrInjectorVersion     = "0.26.0-1"
+	emrJavaVersion         = "1.42.2-1"
+	emrAgentVersion        = "7.58.2-1"
+	commandTimeoutDuration = 10 * time.Second
+)
+
+var (
+	emrInfoPath        = "/mnt/var/lib/info"
+	tracerEnvConfigEmr = []common.InjectTracerConfigEnvVar{
+		{
+			Key:   "DD_DATA_JOBS_ENABLED",
+			Value: "true",
+		},
+		{
+			Key:   "DD_INTEGRATIONS_ENABLED",
+			Value: "false",
+		},
+		{
+			Key:   "DD_DATA_JOBS_COMMAND_PATTERN",
+			Value: ".*org.apache.spark.deploy.*",
+		},
+		{
+			Key:   "DD_SPARK_APP_NAME_AS_SERVICE",
+			Value: "true",
+		},
+		{
+			Key:   "DD_INJECT_FORCE",
+			Value: "true",
+		},
+	}
+)
+
+type emrInstanceInfo struct {
+	IsMaster        bool   `json:"isMaster"`
+	InstanceGroupID string `json:"instanceGroupId"`
+}
+
+type cluster struct {
+	Name string `json:"Name"`
+}
+
+type emrResponse struct {
+	Cluster cluster `json:"cluster"`
+}
+
+type extraEmrInstanceInfo struct {
+	JobFlowID    string `json:"jobFlowId"`
+	ReleaseLabel string `json:"releaseLabel"`
+}
+
+// SetupEmr sets up the DJM environment on EMR
+func SetupEmr(s *common.Setup) error {
+
+	s.Packages.Install(common.DatadogAgentPackage, emrAgentVersion)
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaVersion)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+	s.Config.DatadogYAML.Hostname = hostname
+	s.Config.DatadogYAML.DJM.Enabled = true
+
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
```

Comment:
nit: add a comment about this parameter, since its reason is not obvious

```suggestion
    // Ensure tags are always attached with the metrics
	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
```